modules/packages/LinearAlgebra.chpl

```diff
@@ -1757,6 +1757,261 @@ proc svd(A: [?Adom] ?t) throws
   return (u, s, vt);
 }
 
+pragma "no doc" 
+proc svd(A : [?Adom] ?eltType, iterations : int = 30, eps : real = 1e-8)
+  throws
+  where isRealType(eltType) && !usingLAPACK && isDenseDom(Adom) 
+    && Adom.rank == 2
+{
+  var (m,n) = Adom.shape;
+  ref U = A.reindex(0..#m, 0..#n);
+  var l, nm : int = 0;
+  var flag : bool;
+  var anorm, c, f, g, h, s, scale, x, y, z : eltType = 0.0;
+  var rv1 = Vector(0..#n, eltType);
+
+  var W = Vector(0..#n, eltType);
+  var V = Matrix(0..#n, eltType);
+
+  for i in 0..<n {
+    l = i+2;
+    rv1[i] = scale*g;
+    g = 0.0;
+    s = 0.0;
+    scale = 0.0;
+    
+    if i < m {
+
+      forall k in i..<m with (+ reduce scale) do scale += abs(U[k,i]);
```

Comment:
Can this be written as something like this? (note: I didn't try this myself)

```suggestion
      scale = + reduce (abs(U[1..<m, i]));
```

Reply:
This applies to a bunch of reductions below as well.

Reply:
I did that because Damian mentioned something about in the mailing lists - explicit forall loops performing better for reductions than array slicing. I'll double check the emails and perform some local testing as well to confirm